modules/tinymce/src/plugins/anchor/main/ts/core/FilterContent.ts

```diff
@@ -5,27 +5,28 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
-const isAnchorNode = function (node) {
-  return !node.attr('href') && (node.attr('id') || node.attr('name')) && !node.firstChild;
+import { Arr } from '@ephox/katamari';
+import Node from 'tinymce/core/api/html/Node';
+import Editor from 'tinymce/core/api/Editor';
+
+const isAnchorNode = (node: Node) => {
+  return !node.attr('href') && (node.attr('id') || node.attr('name'));
 };
 
-const setContentEditable = function (state) {
-  return function (nodes) {
-    for (let i = 0; i < nodes.length; i++) {
-      if (isAnchorNode(nodes[i])) {
-        nodes[i].attr('contenteditable', state);
-      }
+const setContentEditable = (state: string | null) => (nodes: Node[]) =>
+  Arr.each((nodes), (node) => {
```

Comment:
This was likely intentionally not using `Arr` as that will increase the code size, since nothing else in the plugin uses `Arr`. It could probably use the `Tools.each` API from core instead, as that wouldn't pull in a new module.

Reply:
Also no need for the parentheses around `(nodes)` here.

- [ ] Remove unneeded parentheses

Reply:
Just to add, using `Arr` is probably fine as it's quite small, but that's why we normally avoid using modules from other packages.

Reply:
Yeah, you are right. I also should have remembered node filters use `for` loops and `while` loops directly. Maybe it would be best to set it back to using a `for` loop to stick with convention?

Reply:
Node filters and the like use it for a different reason (performance) as it has less overhead since it doesn't have to do function calls/add items to the stack/etc...

Reply:
Anyways, I think it's fine to leave `Arr` here.